scripts/dts/edtlib.py

```diff
@@ -171,11 +187,13 @@ def _init_compat2binding(self, bindings_dirs):
 
         self._compat2binding = {}
         for binding_path in self._binding_paths:
-            compat = _binding_compat(binding_path)
-            if compat in dt_compats:
-                binding = _load_binding(binding_path)
-                self._compat2binding[compat, _binding_bus(binding)] = \
-                    (binding, binding_path)
+            compat = None
+            with open(binding_path, encoding="utf-8") as binding:
+                for line in binding:
+                    match = any(compat in line for compat in dt_compats)
```

Comment:
Can do

```python
regex = "|".join(re.escape(compat) for compat in dt_compats)
...
if not re.search(regex, binding.read()):
    continue

# File has one of the strings in it
...
```

to skip files that have none of the compatible strings in them.

The code should also work if the string just happens to appear in the file, e.g. in a comment.

Would be nice if we could avoid adding more ad-hoc parsing though.